cmd/portal/send.go

```diff
@@ -40,30 +49,26 @@ var sendCmd = &cobra.Command{
 		}
 		defer logFile.Close()
 
-		handleSendCommand(args)
+		handleSendCommand(args, cmd.Flag("relay").Changed)
 		return nil
 	},
 }
 
-// Set flags.
-func init() {
-	// Add subcommand flags (dummy default values as default values are handled through viper)
-	//TODO: refactor into a single flag providing a string
-	sendCmd.Flags().IntP("rendezvous-port", "p", 0, "port on which the rendezvous server is running")
-	sendCmd.Flags().StringP("rendezvous-address", "a", "", "host address for the rendezvous server")
-}
+// ------------------------------------------------------ Handler ------------------------------------------------------
 
 // handleSendCommand is the sender application.
-func handleSendCommand(fileNames []string) {
-	addr := viper.GetString("rendezvousAddress")
-	port := viper.GetInt("rendezvousPort")
+func handleSendCommand(fileNames []string, selfHostedRelay bool) {
 	var opts []senderui.Option
 	ver, err := semver.Parse(version)
 	// Conditionally add option to sender ui
 	if err == nil {
 		opts = append(opts, senderui.WithVersion(ver))
 	}
-	sender := senderui.New(fileNames, fmt.Sprintf("%s:%d", addr, port), opts...)
+	relayAddr := viper.GetString("relay")
+	if selfHostedRelay {
+		opts = append(opts, senderui.WithCopyFlags(map[string]string{"--relay": relayAddr}))
+	}
+	sender := senderui.New(fileNames, relayAddr, opts...)
```

Comment:
I'm curious about an alternative approach for this part. What do you think about gettig the `relayAddr` using `viper.getString("relay")` inside the `copyReceiverCommand` method in senderui?

We could then (if I understand the usage of this correctly) see if has changed from the default using https://pkg.go.dev/github.com/spf13/viper#FlagValue.HasChanged

What do you think?

Reply:
Effectively, we are already doing this right? Check 
 
https://github.com/SpatiumPortae/portal/blob/8deae65d730f515b2fe3a94499d24b152fb65d2c/cmd/portal/send.go#L52

`GetString` only returns the string value associated with the key, not a `FlagValue` so we cannot check if it has changed. From what I can gather we have to get the flag values from the cobra command in order to check if they have changed.

Also, this approach allows us to add more flags in the future if we want to.

Reply:
Yes I saw that we already are doing that. I just hoped there'd be a way to not pass arounf:so much information if it can be gotten where it is needed.

Reply:
Hmm I see, we could perhaps directly access it and check if the relay address differs from the default one, But I do not really know how that would interact with user configs. IMO I think it is fine passing it as an option, what do you think?

Reply:
I think it's fine too -- although it might be worth it to just check how this interacts with the changes in #50. I have a feeling that changing the `relay` in the config file to something other than the default would not give you the copy command with the appropriate flags.